src/spec.c

```diff
@@ -1913,6 +1913,10 @@ int IndexSpec_UpdateDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString
   }
 
   if (rv != REDISMODULE_OK) {
+    // if a document did not load properly, it is deleted
+    // to prevent mismatch of index and hash
+    DocTable_DeleteR(&spec->docs, key);
```

Comment:
Does the test cover this part?

Reply:
No.
On hash it cannot be tested b/c the only way to fail is if the key does not exist.
On JSON it can fail so it should be a fairly simple test.

Reply:
So lets test this path on json

Reply:
Done